kin-sdk/KinClient.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Net.Http;
+using kin_base;
+
+namespace kin_sdk
+{
+    public class KinClient
+    {
+        private const string DefualtAppId = "anon";
+        private readonly TimeSpan TransactionTimeout = TimeSpan.FromSeconds(30);
+        private Server server;
+        internal readonly TransactionSender transactionSender;
+        internal readonly AccountInfoRetriver accountInfoRetriver;
+        internal readonly GeneralBlockchainInfoRetriever generalBlockchainInfoRetriever;
+        public readonly Environment environment;
+        public readonly string appId;
+        public readonly IKeyStoreProvider keyStore;
+
+        /// <summary>
+        /// Build a KinClient object
+        /// </summary>
+        /// <param name="environment">The environment to connect to</param>
+        /// <param name="keyStore">An implementation of the IKeyStoreProvider interface</param>
+        /// <param name="appId">A 3/4 charcter string (letters/digits) which represent the application id to add to each transaction</param>
+        /// <param name="httpClient">Optional httpClient to use to communicate with the blockchain</param>
+        public KinClient(Environment environment, IKeyStoreProvider keyStore, string appId = DefualtAppId, HttpClient httpClient=null)
+        {
+            this.environment = environment ?? throw new ArgumentNullException("Environemnt");
+            this.keyStore = keyStore ?? throw new ArgumentNullException("Keystore");
```

Comment:
same applies for the above

this.keyStore = keyStore ?? throw new ArgumentNullException(nameof(keyStore));